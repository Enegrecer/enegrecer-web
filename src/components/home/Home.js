import React from 'react';
import Estatistica from '../estatistica/Estatistica';
import { withHistoryPropTypes } from './sharedPropTypes';
import './css/home.css';

const Home = () => (
  <div className="row">
    <div className="col s12 m6" id="banner">
      <div className="AT-QUANDO-VOC-VAI">
      Até quando você vai se esconder?
      </div>
    </div>
    <Estatistica />
  </div>
);

Home.propTypes = withHistoryPropTypes();

export default Home;
