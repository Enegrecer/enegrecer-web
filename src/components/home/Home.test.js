import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Home from './Home';

describe('Home', () => {
  const historyMock = { push: jest.fn() };
  const wrapper = shallow(<Home history={historyMock} />);

  it('possui a mesma estrutura', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('tem um menu', () => {
    expect(wrapper.find('Menu').exists()).toBeTruthy();
  });

  it('tem um rodapÃ©', () => {
    expect(wrapper.find('Footer').exists()).toBeTruthy();
  });

  it('tem links sociais', () => {
    const social = wrapper.find('Social');

    expect(social.props().networks[0].title).toEqual('Facebook');
    expect(social.props().networks[1].title).toEqual('Twitter');
    expect(social.props().networks[2].title).toEqual('Instagram');
  });
});
