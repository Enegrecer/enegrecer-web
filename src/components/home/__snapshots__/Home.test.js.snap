// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home possui a mesma estrutura 1`] = `
<div
  className="row"
>
  <div
    className="col s12 m6"
    id="banner"
  >
    <div
      className="AT-QUANDO-VOC-VAI"
    >
      Até quando você vai se esconder?
    </div>
  </div>
  <Estatistica />
</div>
`;
